Jenkinsfile

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env groovy
+
+/**
+ * Build Contiki Applications
+ *
+ */
+
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '5')),
+    parameters([
+        stringParam(defaultValue: '26', description: 'Set 6LoWPAN Channel', name: 'CHANNEL'),
+        stringParam(defaultValue: '0xabcd', description: 'Set 6LoWPAN PAN ID', name: 'PAN_ID'),
+        stringParam(defaultValue: '1', description: 'Set Serial port to use for console', name: 'PADS'),
+    ])
+])
+
+node('docker && imgtec') {
+    def docker_image
+    docker_image = docker.image "imgtec/creator-builder:latest" // TODO for now using the creator one
+    docker_image.inside {
+        stage('Prepare') {
+            checkout([$class: 'GitSCM',
+                userRemoteConfigs: scm.userRemoteConfigs,
+                branches: scm.branches,
+                doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
+                submoduleCfg: scm.submoduleCfg,
+                browser: scm.browser,
+                gitTool: scm.gitTool,
+                extensions: scm.extensions + [
+                    [$class: 'CleanCheckout'],
+                    [$class: 'PruneStaleBranch'],
+                ],
+            ])
+        }
+        stage('Build') {
+            sh 'git submodule init "dev/ca8210" && git submodule update'
```

Comment:
What is this for? Either way if you need to call a git command then you should be using the jenkins git or checkout commands instead i.e. should go with the prepare stuff above. Is the submodule already there or are you adding it? If its there already then the checkout command above can be configured to do this for you.